src/DomainServices/Services/Interfaces/ICustomerService.cs

```diff
@@ -1,22 +1,24 @@
 using System;
 using System.Collections.Generic;
-using DesafioWarren.WebAPI.Models;
+using DomainModels;
 
-namespace DesafioWarren.WebAPI.Services;
+namespace DomainServices.Services;
 
 public interface ICustomerService
-{    
-    bool CreateCustomer(Customer customer);
+{
+    (bool isValid, string message) CreateCustomer(Customer customer);
 
     IList<Customer> GetCustomers();
 
-    Customer GetCustomerByEmail(string Email);
+    bool CustomerForCpfAlreadyExists(string cpf);
+
+    bool CustomerForEmailAlreadyExists(string email);
 
     Customer GetCustomerById(Guid CustomerId);
 
     Customer GetCustomerByName(string FullName);
 
     Customer UpdateCustomer(Customer customer);
 
-    bool ExcludeCustomer(Guid id);
+    bool DeleteCustomer(Guid id);
```

Comment:
Como já estamos no contexto de Customer, é redundante especificar que está deletando um Customer, o mesmo vale para os outros métodos

```suggestion
    bool Delete(Guid id);
```